include/open62541/server.h

```diff
@@ -1839,6 +1839,10 @@ UA_StatusCode UA_EXPORT UA_THREADSAFE
 UA_Server_getNamespaceByIndex(UA_Server *server, const size_t namespaceIndex,
                               UA_String *foundUri);
 
+/* Add a new typeArray to the server. */
```

Comment:
This needs documentation whether we make a deep copy of the array.
Otherwise we can add a flag to indicate that an array is always accessible and does not need to be copied. That means:

- the array outlives the lifetime of the server
- the array can be accessed from any thread (not on the stack)
- the array is not cleaned up automatically

Given all that it could be safer to always make a deep copy.
What do you think?

The custom types are part of the visible server config.
Do we add to the server or to the config? The api should be clear about that.
Will serverconfig _clear delete the array?

how do we handle server-specific namespace-index?
Where will the rewriting of the datatype namespace index take place?